### PR TITLE
[R4R]request block concurrently

### PR DESCRIPTION
**Pull request by byteflyfunny**
# Goals of PR

Core changes:

- Describe changes here

Notes:

- Write notes here

Related Issues:

- Link issues here

### CODE REVIEW

File: l2geth/rollup/sync_service.go [hunk 1]
```diff
@@ -1528,19 +1533,71 @@ func (s *SyncService) syncTransactions(backend Backend) (*uint64, error) {
...
+		return fmt.Errorf("offset %d is small than start %d", offset, start)
+	}
+	wg.Add(int(end - start))
+
+	for i := start; i < end; i++ {
+		go func(index uint64) {
+			defer wg.Done()
+			tx, err := s.client.GetTransaction(index, backend)
+			if err != nil {
+				returnErr = err
```
Comment (HaoyangLiu) [inline l2geth/rollup/sync_service.go]: concurrent write risk

Comment (byteflyfunny) [reply l2geth/rollup/sync_service.go]: done

**Comment by HaoyangLiu** [comment]
Do you think it would be better if `dtl` can support getting batch `txs`.

**Comment by byteflyfunny** [comment]
> Do you think it would be better if `dtl` can support getting batch `txs`.

We plan to proceed this way in the future.

File: l2geth/rollup/sync_service.go [hunk 2]
```diff
@@ -1528,19 +1533,92 @@ func (s *SyncService) syncTransactions(backend Backend) (*uint64, error) {
 	return index, nil
 }
 
+func (s *SyncService) getTransactions(start, end, offset uint64, txs []*types.Transaction, backend Backend) error {
+	var wg sync.WaitGroup
+	var returnErr error
```
Comment (HaoyangLiu) [inline l2geth/rollup/sync_service.go:1538]: I think the `returnErr` might not be necessary.

Comment (byteflyfunny) [reply l2geth/rollup/sync_service.go:1538]: return an error to break out of the upper-level for loop. The code has been fixed.

**Review by HaoyangLiu — APPROVED** [review]

**Pull request closed without merging**